wakuv2/waku.go

```diff
@@ -1209,23 +1206,9 @@ func (w *Waku) Start() error {
 					w.onPeerStats(latestConnStatus)
 				}
 
-				if w.cfg.EnableDiscV5 {
-					// Restarting DiscV5
-					if !latestConnStatus.IsOnline && isConnected {
-						w.logger.Info("Restarting DiscV5: offline and is connected")
-						isConnected = false
-						w.node.DiscV5().Stop()
-					} else if latestConnStatus.IsOnline && !isConnected {
-						w.logger.Info("Restarting DiscV5: online and is not connected")
-						isConnected = true
-						if w.node.DiscV5().ErrOnNotRunning() != nil {
-							err := w.node.DiscV5().Start(w.ctx)
-							if err != nil {
-								w.logger.Error("Could not start DiscV5", zap.Error(err))
-							}
-						}
-					}
-				}
+				w.ConnectionChanged(connection.State{
```

Comment:
We will have to consider if discv5 is enabled or not, and if not then not do anything within handler for this connectionChangedEvent .  Otherwise this code will fail for lightNodes.

Reply:
Hmm, If i see https://github.com/status-im/status-go/blob/94e5431ea311079cc5cdd0e185b646497e134a7f/wakuv2/waku.go#L1597 it looks like there is no check for discv5. 
Am i missing something?

Reply:
You're right. I did not check the correct spot! Fixing this in a commit